saleor/webhook/transport/metrics.py

```diff
@@ -50,11 +59,31 @@
     bucket_boundaries=BODY_SIZE_BUCKETS,
 )
 
+METRIC_EXTERNAL_REQUEST_FIRST_ATTEMPT_DELAY = meter.create_metric(
+    "saleor.external_request.async.first_attempt_delay",
+    scope=Scope.CORE,
+    type=MetricType.HISTOGRAM,
+    unit=Unit.MILLISECOND,
+    description="Delay of the first delivery attempt for async webhook.",
+)
+
 
 def record_external_request(
-    target_url: str, webhook_response: WebhookResponse, payload_size: int
+    event_type: str,
+    target_url: str,
+    webhook_response: WebhookResponse,
+    payload_size: int,
+    sync: bool,
+    app: App | None,
 ) -> None:
-    attributes = {server_attributes.SERVER_ADDRESS: urlparse(target_url).hostname or ""}
+    attributes = {
+        server_attributes.SERVER_ADDRESS: urlparse(target_url).hostname or "",
+        saleor_attributes.SALEOR_WEBHOOK_EVENT_TYPE: event_type,
+        saleor_attributes.SALEOR_WEBHOOK_EXECUTION_MODE: "sync" if sync else "async",
+    }
+    if app:
```

Comment:
Do you actually need `app: App | None` and `if app:`? 
You can't have the webhook, without the `app` (so you can't also have `webhook.target_url`), so maybe we can simplify the typing, and dropping the `if` condition?